microsoft/vscode
# Apply enable-proposed-api Changes in argv.json with Window Reload

Updated the `updateEnabledApiProposals` function to be asynchronous. It now checks the `argv.json` file for the latest "enable-proposed-api" field and provides an up-to-date response.

Feature: #224695 
Closes #224695

### Testing Process

1. **Enabling Proposed API for Specific Extension**
   - Open the `argv.json` file.
   - Add or modify the line to include `"enable-proposed-api": ["<extension-id>"]` (replace `<extension-id>` with the actual ID of the extension you want to test).
   - Save the file and Click the "Reload Window" button.
   - Verify that the proposed API features for the specified extension are enabled after the reload by checking the availability of any proposed API functionalities provided by that extension.

2. **Disabling Proposed API for Specific Extension**
   - Open the `argv.json` file again.
   - Remove or comment out the `"enable-proposed-api": ["<extension-id>"]` line.
   - Save the file and Click the "Reload Window" button.
   - Verify that the proposed API features for the specified extension are disabled after the reload by checking that the proposed API functionalities provided by that extension are no longer available.
